DataFormats/Detectors/TRD/include/DataFormatsTRD/TrapConfig3.h

```diff
@@ -0,0 +1,619 @@
+  unsigned int getShift() { return mShift; }
+  uint32_t getMask() { return mMask; }
+  uint32_t getMax() { return mMax; }
+  bool getIgnoreChange() { return mIgnoreChange; }
+
+  void setName(const std::string name) { mName = name; }
+  void setAddr(const uint32_t addr) { mAddr = addr; }
+  void setNbits(const uint32_t bits) { mNbits = bits; }
+  void setBase(const uint32_t base) { mBase = base; }
+  void setWordNumber(const uint32_t wordnum) { mWordNumber = wordnum; }
+  void setDataWordNumber(const uint32_t datawordnum) { mDataWordNumber = datawordnum; }
+  void setShift(const uint32_t shift) { mShift = shift; }
+  void setMask(const uint32_t mask) { mMask = mask; }
+  void setMax(uint32_t max) { mMax = pow(2, max) - 1; }
+  void setIgnoreChange(const uint32_t ignorechange) { mIgnoreChange = ignorechange; }
+
+  void logTrapRegInfo(); // output the contents to log info
+
+ private:
+  TrapRegInfo(const TrapRegInfo& rhs);
+  TrapRegInfo& operator=(const TrapRegInfo& rhs);
+
+  // fixed properties of the register
+  // which do not need to be stored on a per mcm basis
+  std::string mName;        //!< Name of the register
+  uint16_t mAddr;           //!< Address in GIO of TRAP
+  uint16_t mNbits;          //!< Number of bits, from 1 to 32
+  uint32_t mBase;           //!< base of this registers block, i.e. TFL?? will have 0 and is in the range [0,kTrapRegistersSize]
+  uint32_t mWordNumber;     //!< word number offset, of size Nbits, in the block of registers
+  uint32_t mDataWordNumber; //!< offset, into "compressed" 32 bit words for the 32 bit word containing this register, offset from the base;
+  uint32_t mMask;           //!< mask to extract register from the word identified by WordNumber
+  uint32_t mMax;            //!< max is not the same as the mask, some values come in as 15 bit mask but are only 12 or 13 bits for max, IRQ values for example
+  uint32_t mShift;          //!< shift to extract the register
+  bool mIgnoreChange;       //!< we are not concerned with this value changing for purposes of the differential comparison.
+  ClassDefNV(TrapRegInfo, 1);
+};
+
+class TrapConfig3
+{
+  // class that is actually stored in ccdb.
+  // it holds a compressed version of what comes in the config events
+ public:
+  TrapConfig3();
+  ~TrapConfig3() = default;
+
+  // get a config register value by index, addr, and name, via mcm index
+  uint32_t getRegisterValue(uint32_t regidx, int mcmidx);
+  bool setRegisterValue(uint32_t data, uint32_t regidx, int mcmidx);
+  uint32_t getRegisterValueByIdx(uint32_t regidx, int mcmidx);
+  bool setRegisterValueByIdx(uint32_t data, uint32_t regidx, int mcmidx);
+  uint32_t getRegisterValueByAddr(uint32_t addr, int mcmidx);
+  bool setRegisterValueByAddr(uint32_t data, uint32_t addr, int mcmidx);
+  uint32_t getRegisterValueByName(const std::string& name, int mcmidx);
+  bool setRegisterValueByName(uint32_t data, const std::string& regname, int mcmidx);
```

Comment:
Are all of these methods actually used? I would assume that the user should always refer to the registers using the enum values above (e.g. kADCMSK), and the name and addresses in MCM GIO address space are mostly a convencience for the reader.

Also, what is the difference between getRegisterValue() and getRegisterValueByIdx()?